### PR TITLE
Update the codegen to say returning Promise.

### PR DESCRIPTION
**Pull request by jmuk**
Related to googleapis/gax-nodejs#39

### CODE REVIEW

File: src/test/java/com/google/api/codegen/testdata/nodejs_doc_main_library.baseline [hunk 1]
```diff
@@ -398,12 +398,12 @@ LibraryServiceApi.prototype.getShelf = function(
...
  *   The third item will be set if the response contains the token for the further results
  *   and can be reused to `pageToken` field in the options in the next request.
- * @returns {Stream|gax.EventEmitter}
+ * @returns {Stream|Promise}
  *   An object stream which emits an object representing
  *   [Shelf]{@link Shelf} on 'data' event.
  *   When the callback is specified or streaming is suppressed through options,
- *   it will return an event emitter to handle the call status and the callback
- *   will be called with the response object.
+ *   it will return a a promise that resolves to the response object. The promise
```
Comment (landrito) [inline src/test/java/com/google/api/codegen/testdata/nodejs_doc_main_library.baseline]: Looks like a typo here causing CI errors.
s/a a promise/a promise

File: src/main/java/com/google/api/codegen/nodejs/NodeJSGapicContext.java [hunk 2]
```diff
@@ -318,13 +318,13 @@ private String returnTypeComment(Method method, MethodConfig config) {
...
-          + "\n@returns {Stream|gax.EventEmitter}\n"
+          + "\n@returns {Stream|Promise}\n"
           + "  An object stream which emits "
           + resourceTypeName
           + " on 'data' event.\n"
           + "  When the callback is specified or streaming is suppressed through options,\n"
-          + "  it will return an event emitter to handle the call status and the callback\n"
-          + "  will be called with the response object.";
+          + "  it will return a promise that resolves to the response object. The promise\n"
+          + "  has cancel method which cancels the ongoing API call.";
```
Comment (landrito) [inline src/main/java/com/google/api/codegen/nodejs/NodeJSGapicContext.java]: nit: s/has cancel method/has a cancel method

File: src/main/java/com/google/api/codegen/nodejs/NodeJSGapicContext.java [hunk 3]
```diff
@@ -345,18 +345,9 @@ private String returnTypeComment(Method method, MethodConfig config) {
...
-    if (config.isBundling()) {
-      returnMessage +=
-          " When isBundling: false is specified in the options, it still returns\n"
-              + "  a gax.BundleEventEmitter but the API is immediately invoked, so it behaves same\n"
-              + "  as a gax.EventEmitter does.";
-    }
-    return callbackMessage + "\n" + returnMessage;
+    return callbackMessage
+        + "\n@returns {Promise} - The promise which resolves to the response object.\n"
+        + "  The promise has cancel method which cancels the ongoing API call.";
```
Comment (landrito) [inline src/main/java/com/google/api/codegen/nodejs/NodeJSGapicContext.java]: nit: s/has cancel method/has a cancel method

File: src/main/java/com/google/api/codegen/nodejs/NodeJSGapicContext.java (hunk 2 above)
Comment (jmuk) [inline src/main/java/com/google/api/codegen/nodejs/NodeJSGapicContext.java]: Here I wanted to say the promise has a method, and the name of the method is "cancel", and 'a cancel method' sounds a bit unclear to me about the method name.

Changed as `has a method named "cancel" which...`

File: src/main/java/com/google/api/codegen/nodejs/NodeJSGapicContext.java (hunk 3 above)
Comment (jmuk) [inline src/main/java/com/google/api/codegen/nodejs/NodeJSGapicContext.java]: same.

File: src/test/java/com/google/api/codegen/testdata/nodejs_doc_main_library.baseline (hunk 1 above)
Comment (jmuk) [inline src/test/java/com/google/api/codegen/testdata/nodejs_doc_main_library.baseline]: Done.

**Comment by landrito** [comment]
LGTM

**Pull request merged**